src/uu/mktemp/src/mktemp.rs

```diff
@@ -281,7 +281,11 @@ impl Params {
             .join(prefix_from_template)
             .display()
             .to_string();
-        if options.treat_as_template && prefix.contains(MAIN_SEPARATOR) {
+
+        // Check that the prefix is valid.
+        let prefix_of_template = Path::new(prefix_from_template).display().to_string();
```

Comment:
```suggestion
        let prefix_of_template = Path::new(prefix_from_template).to_string_lossy();
```

This is a slightly shorter way to do the same thing.

Reply:
Works perfectly, thanks for the suggestion